File: benchmarks/ListAlloc/ListAlloc.java
import java.util.Arrays;

public class ListAlloc {
    public static void allocate() {
        int arr[] = new int[1000];
        Arrays.fill(arr, 0);
    }

    public static void main(String args[]) {
        long[] results = new long[125];

        for (int i=0; i < 125; ++i) {
            long startTime = System.nanoTime() / 1000000;
            allocate();
            long stopTime = System.nanoTime() / 1000000;
            results[i] = stopTime - startTime;
        }

        double mean = 0.0;
        for (long a : results)
            mean += a;
        mean /= 125;

        System.out.println("Mean time: " + mean + "ms");

        Arrays.sort(results);
        if (results.length % 2 == 0)
            System.out.println("Median time: " + (results[(results.length / 2) - 1] + results[results.length / 2]) / 2.0 + "ms");
        else
            System.out.println("Median time: " + results[results.length / 2] + "ms");

        double temp = 0.0;
        for (long a : results)
            temp += (a - mean) * (a - mean);
        double stddev = Math.sqrt(temp / 125);

        System.out.println("Stddev: " + stddev + "ms");
    }
}

File: benchmarks/ListAlloc/list_alloc.js
function allocate() {
    data = new Array(1000).fill(0);
    return data;
}

results = [];
for (var i = 0; i < 125; ++i) {
    var t0 = performance.now();
    allocate();
    var stop = performance.now() - t0;
    results.push(stop);
}

mean = 0;
for (var i = 0; i < 125; ++i) {
    mean += results[i];
}
mean /= 125;

console.log(`Mean time ${mean} ms`);

results.sort();

if (results.length % 2 == 0)
    console.log(`Median time: ${(results[Math.floor((results.length / 2) - 1)] + results[Math.floor(results.length / 2)]) / 2.0} ms`);
else
    console.log(`Median time: ${results[Math.floor(results.length / 2)]} ms`);

temp = 0;
for (var i = 0; i < 125; ++i) {
    temp += (results[i] - mean) * (results[i] - mean);
}
stddev = Math.sqrt(temp / 125);

console.log(`Stddev: ${stddev} ms`);